test/run_test.py

```diff
@@ -313,6 +313,13 @@ def run_test(test_module, test_directory, options, launcher_cmd=None, extra_unit
     if extra_unittest_args:
         assert isinstance(extra_unittest_args, list)
         unittest_args.extend(extra_unittest_args)
+
+    # If using pytest, replace -f with equivalent -x
+    if options.pytest:
+        for idx, arg in enumerate(unittest_args):
+            if arg == '-f':
+                unittest_args[idx] = '-x'
```

Comment:
Wouldn't list comprehension make it a bit more concise here?
```suggestion
        unittest_args = [arg if arg != '-f' else '-x' for arg in unittest_args]
```